httpserver/service_node.cpp

```diff
@@ -779,93 +718,75 @@ void ServiceNode::ping_peers_tick() {
         return;
     }
 
+    auto now = std::chrono::steady_clock::now();
+
     // Check if we've been tested (reached) recently ourselves
-    reach_records_.check_incoming_tests(all_stats_.get_reset_time());
+    reach_records_.check_incoming_tests(now);
 
     if (this->status_ == SnodeStatus::DECOMMISSIONED) {
         OXEN_LOG(debug, "Skipping this round of peer testing (decommissioned)");
         return;
     }
 
-    /// We always test one node already known to be offline
-    /// plus one random other node (could even be the same node)
+    /// We always test nodes due to be tested plus one general, non-failing node.
 
-    const auto random_node = swarm_->choose_funded_node();
+    auto to_test = reach_records_.get_failing(*swarm_, now);
+    if (auto rando = reach_records_.next_random(*swarm_, now))
+        to_test.emplace_back(std::move(*rando), 0);
 
-    if (random_node) {
-
-        if (random_node == our_address_) {
-            OXEN_LOG(trace, "Would test our own node, skipping");
-        } else {
-            OXEN_LOG(trace, "Selected random node for testing: {}",
-                     (*random_node).pub_key_hex());
-            test_reachability(*random_node);
-        }
-    } else {
-        OXEN_LOG(trace, "No nodes to test for reachability");
-    }
+    if (to_test.empty())
```

Comment:
is this check for empty required? you use a for each below.

Reply:
It wasn't, but now it guards whether we trace ("no nodes to test") or debug ("x nodes to test"), and the unneeded return is gone.